### PR TITLE
[CELEBORN-50] Channel inActive may cause new client use old stream id to fetch data cause IllegalStateException.

### PR DESCRIPTION
**Pull request by FMX**
# [CELEBORN-50][BUG] Fix fetch chunk illegal state exception.

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: client/src/main/java/org/apache/celeborn/client/read/Replica.java [hunk 1]
```diff
@@ -58,7 +58,12 @@ class Replica {
   public synchronized TransportClient getOrOpenStream() throws IOException, InterruptedException {
     if (client == null || !client.isActive()) {
       client = clientFactory.createClient(location.getHost(), location.getFetchPort());
+      OpenStream openBlocks =
+          new OpenStream(shuffleKey, location.getFileName(), startMapIndex, endMapIndex);
+      ByteBuffer response = client.sendRpcSync(openBlocks.toByteBuffer(), timeoutMs);
+      streamHandle = (StreamHandle) Message.decode(response);
```
[the remark this answers is not in the document]
Comment (FMX) [reply client/src/main/java/org/apache/celeborn/client/read/Replica.java]: It is updated.

File: client/src/main/java/org/apache/celeborn/client/read/Replica.java [hunk 2]
```diff
@@ -58,16 +58,22 @@ class Replica {
   public synchronized TransportClient getOrOpenStream() throws IOException, InterruptedException {
     if (client == null || !client.isActive()) {
```
Comment (AngersZhuuuu) [inline client/src/main/java/org/apache/celeborn/client/read/Replica.java:59]: log for client is not active?

**Comment by AngersZhuuuu** [comment]
pr totle 
```
Channel inActive may cause new client use old stream id to fetch data cause IllegalStateException
```

File: client/src/main/java/org/apache/celeborn/client/read/Replica.java (hunk 2 above)
Comment (AngersZhuuuu) [reply client/src/main/java/org/apache/celeborn/client/read/Replica.java:59]:
```
if (client != null) {
        logger.error(
            "Current channel from "
                + client.getChannel().localAddress()
                + " to "
                + client.getChannel().remoteAddress()
                + " for "
                + this
                + " is not active.");
      }
```

Comment (FMX) [reply client/src/main/java/org/apache/celeborn/client/read/Replica.java:62]: How about now ?

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

**Pull request merged**